pkg/packetbrokeragent/is.go

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packetbrokeragent
+
+import (
+	"context"
+	"time"
+
+	"github.com/bluele/gcache"
+	"go.thethings.network/lorawan-stack/v3/pkg/cluster"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"google.golang.org/grpc"
+)
+
+var (
+	cacheSize = 4096
+	cacheTTL  = 24 * 60 * 60 * time.Second // 24 hour
+)
```

Comment:
Gave a look into the Grafana stats and I'm questioning myself regarding the cache size, the amount of gateways connected is a lot larger than this but I'm assuming that there are a few replicas of the PBA on production, so the cache size might be appropriate but I wanted to confirm that with you @adriansmares.